device_backends/include/NumericAddressedBackendRegisterAccessor.h

```diff
@@ -78,56 +80,136 @@ namespace ChimeraTK {
-      _dataConverter.template vectorToCooked<UserType>(itsrc, itsrc + _buffer_2D[0].size(), _buffer_2D[0].begin());
+      callForRawType(_registerInfo->getDataDescriptor().rawDataType(), [this](auto t) {
+        typedef decltype(t) RawType;
+        auto itsrc = (RawType*)_rawAccessor->begin(_startAddress);
+        _dataConverter.template vectorToCooked<UserType>(itsrc, itsrc + _buffer_2D[0].size(), _buffer_2D[0].begin());
+      });
     }
 
     template<typename UserType, typename DataConverterType, bool isRaw>
     void NumericAddressedPrePostActionsImplementor<UserType, DataConverterType, isRaw>::doPreWrite() {
       if(_isNotWriteable) {
-        throw ChimeraTK::logic_error("NumericAddressedBackend: "
-                                     "Writeing to a non-writeable register is not allowed"
-                                     /*"(Register name: " + _registerPathName + ")*/ ".");
-      }
-      auto itsrc = _rawAccessor->begin(_startAddress);
-      for(auto itdst = _buffer_2D[0].begin(); itdst != _buffer_2D[0].end(); ++itdst) {
-        *itsrc = _dataConverter.template toRaw<UserType>(*itdst);
-        ++itsrc;
+        throw ChimeraTK::logic_error(
+            "NumericAddressedBackend: Writeing to a non-writeable register is not allowed (Register name: " +
+            _registerInfo->getRegisterName() + ").");
       }
+      callForRawType(_registerInfo->getDataDescriptor().rawDataType(), [this](auto t) {
+        typedef decltype(t) RawType;
+        auto itsrc = (RawType*)_rawAccessor->begin(_startAddress);
+        for(auto itdst = _buffer_2D[0].begin(); itdst != _buffer_2D[0].end(); ++itdst) {
+          *itsrc = _dataConverter.template toRaw<UserType>(*itdst);
+          ++itsrc;
+        }
+      });
     }
 
     // special implementations for int32 raw
     template<typename DataConverterType>
     void NumericAddressedPrePostActionsImplementor<int32_t, DataConverterType, true>::doPostRead() {
-      if(!_rawAccessor->isShared) {
-        _buffer_2D[0].swap(_rawAccessor->rawDataBuffer);
-      }
-      else {
-        auto itsrc = _rawAccessor->begin(_startAddress);
-        auto itdst = _buffer_2D[0].begin();
-        memcpy(&(*itdst), &(*itsrc), _buffer_2D[0].size() * sizeof(int32_t));
-      }
+      auto itsrc = _rawAccessor->begin(_startAddress);
+      auto itdst = _buffer_2D[0].begin();
+      memcpy(&(*itdst), &(*itsrc), _buffer_2D[0].size() * sizeof(int32_t));
     }
 
     template<typename DataConverterType>
     void NumericAddressedPrePostActionsImplementor<int32_t, DataConverterType, true>::doPreWrite() {
-      if(!_rawAccessor->isShared) {
-        _buffer_2D[0].swap(_rawAccessor->rawDataBuffer);
-      }
-      else {
-        auto itdst = _rawAccessor->begin(_startAddress);
-        auto itsrc = _buffer_2D[0].begin();
-        memcpy(&(*itdst), &(*itsrc), _buffer_2D[0].size() * sizeof(int32_t));
-      }
+      auto itdst = _rawAccessor->begin(_startAddress);
```

Comment:
Hm, I guess for writing we could just get a fresh copy of _buffer2D[0].data(). that's all we need for the write operation under the hood. No need to copy the data in case of a raw accessor.

Reply:
This is only for raw transfers. This should not be used frequently (maybe legacy servers still use it, but it should be a small fraction). If we want to keep it, we need to also check whether the start address and length matches, because it also won't work for unaligned access.

Reply:
Actually, the swap is no longer implementable. The underlying data type changed from std::vector<int32_t> to std::vector<int8_t>, so swap would only be possible for int8_t raw types - which we rarely use (so far not at all).

Reply:
The swap would be implementable if we template it to the user type. The user type size must be a multiple of the alignment parameter anyway, so we can check this.
For iterating the container in bytes we could take the raw buffer and reinterpret_cast it to char*, and "iterate" the C-style with pointers. We are in that territory anyway when we are doing these things.

I am thinking about performance-critical applications which skip parts of the abstraction by getting a raw buffer via DMA, and then reinterpret-cast it to a struct with the right memory layout. Except for debugging this is the only legitimate use case of raw transfer I can think of, and here you want to avoid the extra copy. 
One could argue that in this case you can directly interact with the driver, but I think you can still benefit from the device abstraction, memory allocation etc. that DeviceAccess is offering.

Reply:
I just read the comment in NumericAccressedBackendRegisterAccessor. I will resolve this conversation and write there.
The change can stay as it is for now.